migen/build/lattice/icestorm.py

```diff
@@ -82,6 +82,23 @@ def _run_icestorm(build_name, source, yosys_opt, pnr_opt,
 
 
 class LatticeIceStormToolchain:
+    attr_translate = {
+        "keep": ("keep", "true"),
+        "no_retiming": None,  # yosys does not do retiming
+        "async_reg": None,
+
+        # The next 5 attributes are Vivado-specific. Ignore.
```

Comment:
No. Those are not Vivado specific at all. If you can translate them into your toolchain then do so.